File: web/src/api/approval.ts
import { request } from './client'

const API_BASE = '/approval'

// 前端统一使用 camelCase，字段映射在本文件内完成

export interface ApprovalHistoryRecord {
  id: string
  command: string
  normalized?: string
  riskLevel: string
  decision: string
  strategy: string
  durationMs: number
  workingDir?: string
  timestamp: string
}

export interface ApprovalHistoryResult {
  records: ApprovalHistoryRecord[]
  total: number
}

export interface ApprovalStats {
  totalRequests: number
  autoApproved: number
  userApproved: number
  userDenied: number
  riskDistribution: Record<string, number>
  topCommands: { command: string; count: number; riskLevel: string }[]
}

export interface TrustedPattern {
  pattern: string
  count: number
  lastSeen: string
  riskLevel?: string
}

export interface DeniedPattern {
  pattern: string
  count: number
  lastSeen?: string
  riskLevel?: string
}

export interface ApprovalSettings {
  strategy: string
  timeoutStrategy: string
  trustThreshold: number
  enableLearning: boolean
  cliConfirm: boolean
  whitelist: string[]
}

export interface PendingApproval {
  id: string
  command: string
  riskLevel: string
  sessionId?: string
  createdAt: string
  expiresAt?: string
}

// 任意原始记录类型，用于从后端 snake_case 到前端 camelCase 的映射
type AnyRecord = Record<string, any>

// 后端 ByRiskLevel 的 map key 是 RiskLevel（基于 int 的自定义类型），encoding/json
// 会把 key 序列化成数字字符串 "1"/"2"/"3"/"4"。这里归一化成前端使用的语义 key。
function normalizeRiskDistribution(raw: AnyRecord | undefined): Record<string, number> {
  const map: Record<string, number> = {}
  if (!raw) return map
  const numToKey: Record<string, string> = {
    '0': 'low', '1': 'low',
    '2': 'medium',
    '3': 'high',
    '4': 'critical',
  }
  for (const [k, v] of Object.entries(raw)) {
    const key = numToKey[String(k)] || String(k)
    map[key] = (map[key] || 0) + (Number(v) || 0)
  }
  return map
}

// 审批历史：后端返回 { records: [...], total: N }，记录字段为 snake_case
export async function getApprovalHistory(limit: number = 100, offset: number = 0): Promise<ApprovalHistoryResult> {
  const raw = await request<{ records?: AnyRecord[]; total?: number; items?: AnyRecord[] } | AnyRecord[]>(
    `${API_BASE}/history?limit=${limit}&offset=${offset}`,
  )
  // 兼容数组和对象两种返回格式
  const list: AnyRecord[] = Array.isArray(raw)
    ? raw
    : (raw?.records || raw?.items || [])
  const total = Array.isArray(raw) ? raw.length : (raw?.total ?? list.length)
  const records: ApprovalHistoryRecord[] = list.map((r: AnyRecord) => ({
    id: r.id || '',
    command: r.command || '',
    normalized: r.normalized,
    riskLevel: r.risk_level || r.riskLevel || 'low',
    decision: r.decision || '',
    strategy: r.strategy || '',
    durationMs: r.duration_ms ?? r.durationMs ?? 0,
    workingDir: r.working_dir || r.workingDir,
    timestamp: r.timestamp || '',
  }))
  return { records, total }
}

// 审批统计：后端返回 snake_case 字段
export async function getApprovalStats(): Promise<ApprovalStats> {
  const raw = await request<AnyRecord>(`${API_BASE}/stats`)
  return {
    totalRequests: raw.total_requests || raw.totalRequests || 0,
    autoApproved: raw.auto_approved || raw.autoApproved || 0,
    userApproved: raw.user_approved || raw.userApproved || 0,
    userDenied: raw.user_denied || raw.userDenied || 0,
    riskDistribution: normalizeRiskDistribution(raw.by_risk_level || raw.risk_distribution || raw.riskDistribution),
    topCommands: (raw.top_commands || raw.topCommands || []).map((c: AnyRecord) => ({
      command: c.pattern || c.command || '',
      count: c.count || 0,
      riskLevel: c.risk_level || c.riskLevel || 'low',
    })),
  }
}

// 受信任模式：后端返回 { patterns: [...], total: N }，记录字段为 snake_case
export async function getTrustedPatterns(): Promise<TrustedPattern[]> {
  const result = await request<{ patterns?: AnyRecord[]; total?: number }>(`${API_BASE}/patterns/trusted`)
  return (result?.patterns || []).map((p: AnyRecord) => ({
    pattern: p.pattern || '',
    count: p.count || 0,
    lastSeen: p.last_seen || p.lastSeen || '',
    riskLevel: p.risk_level || p.riskLevel,
  }))
}

// 被拒绝模式：后端返回 { patterns: [...], total: N }，记录字段为 snake_case
export async function getDeniedPatterns(): Promise<DeniedPattern[]> {
  const result = await request<{ patterns?: AnyRecord[]; total?: number }>(`${API_BASE}/patterns/denied`)
  return (result?.patterns || []).map((p: AnyRecord) => ({
    pattern: p.pattern || '',
    count: p.count || 0,
    lastSeen: p.last_seen || p.lastSeen,
    riskLevel: p.risk_level || p.riskLevel,
  }))
}

// 待审批：后端返回 { pending: [...], total: N }，记录字段为 snake_case
// sessionId 可选，传入时后端按会话过滤，避免跨会话泄露其他会话的待审批命令。
export async function getPendingApprovals(sessionId?: string): Promise<PendingApproval[]> {
  const url = sessionId
    ? `${API_BASE}/pending?session_id=${encodeURIComponent(sessionId)}`
    : `${API_BASE}/pending`
  const raw = await request<{ pending?: AnyRecord[]; total?: number }>(url)
  const items = raw?.pending || []
  return items.map((p: AnyRecord) => ({
    id: p.id,
    command: p.command || '',
    riskLevel: p.risk_level || 'low',
    sessionId: p.session_id || '',
    createdAt: p.created_at || '',
    expiresAt: p.expires_at || '',
  }))
}

// 审批单个待审批项
export async function resolvePendingApproval(id: string, approved: boolean, reason: string = ''): Promise<void> {
  return request(`${API_BASE}/pending/${id}/resolve`, {
    method: 'POST',
    body: JSON.stringify({ approved, reason }),
  })
}

// 移除受信任模式
export async function removeTrustedPattern(pattern: string): Promise<void> {
  return request(`${API_BASE}/patterns/trusted`, {
    method: 'DELETE',
    body: JSON.stringify({ pattern }),
  })
}

// 清除被拒绝模式
export async function clearDeniedPattern(pattern: string): Promise<void> {
  return request(`${API_BASE}/patterns/denied`, {
    method: 'DELETE',
    body: JSON.stringify({ pattern }),
  })
}

// 获取白名单：后端返回 { patterns: [...] } 或 { whitelist: [...] }
export async function getWhitelist(): Promise<string[]> {
  const result = await request<{ patterns?: string[]; whitelist?: string[] } | string[]>(`${API_BASE}/whitelist`)
  if (Array.isArray(result)) return result
  return result?.patterns || result?.whitelist || []
}

// 添加白名单
export async function addWhitelist(pattern: string): Promise<void> {
  return request(`${API_BASE}/whitelist`, {
    method: 'POST',
    body: JSON.stringify({ pattern }),
  })
}

// 移除白名单
export async function removeWhitelist(pattern: string): Promise<void> {
  return request(`${API_BASE}/whitelist`, {
    method: 'DELETE',
    body: JSON.stringify({ pattern }),
  })
}

// 获取审批设置：后端返回 snake_case 字段
// fallback 值与后端 DefaultConfig / DefaultApprovalConfig 保持一致
export async function getSettings(): Promise<ApprovalSettings> {
  const raw = await request<AnyRecord>(`${API_BASE}/settings`)
  return {
    strategy: raw.strategy || 'smart',
    timeoutStrategy: raw.timeout_strategy ?? raw.timeoutStrategy ?? 'deny',
    trustThreshold: raw.trust_threshold ?? raw.trustThreshold ?? 3,
    enableLearning: raw.enable_learning ?? raw.enableLearning ?? true,
    cliConfirm: raw.cli_confirm ?? raw.cliConfirm ?? false,
    whitelist: raw.whitelist || [],
  }
}

// 保存审批设置：转换为后端 snake_case 字段
export async function saveSettings(settings: ApprovalSettings): Promise<void> {
  return request(`${API_BASE}/settings`, {
    method: 'PUT',
    body: JSON.stringify({
      strategy: settings.strategy,
      timeout_strategy: settings.timeoutStrategy,
      trust_threshold: settings.trustThreshold,
      enable_learning: settings.enableLearning,
      cli_confirm: settings.cliConfirm,
      whitelist: settings.whitelist,
    }),
  })
}

// 清理审批历史
export async function clearHistory(olderThanHours: number): Promise<void> {
  return request(`${API_BASE}/clear-history`, {
    method: 'POST',
    body: JSON.stringify({ older_than_hours: olderThanHours }),
  })
}

File: internal/server/approval.go
package server

import (
	"encoding/json"
	"net/http"
	"strconv"
	"strings"
	"time"

	"github.com/magicwubiao/go-magic/internal/approval"
	appconfig "github.com/magicwubiao/go-magic/pkg/config"
)

func (s *Server) handleApprovalHistory(w http.ResponseWriter, r *http.Request) {
	mgr := s.getApprovalManager()
	if mgr == nil {
		jsonResponse(w, map[string]interface{}{"error": "approval system not available"})
		return
	}

	limitStr := r.URL.Query().Get("limit")
	limit := 20
	if limitStr != "" {
		if n, err := strconv.Atoi(limitStr); err == nil && n > 0 {
			limit = n
		}
	}

	offsetStr := r.URL.Query().Get("offset")
	offset := 0
	if offsetStr != "" {
		if n, err := strconv.Atoi(offsetStr); err == nil && n >= 0 {
			offset = n
		}
	}

	records := mgr.GetHistory(limit, offset)
	total := mgr.HistoryLen()

	jsonResponse(w, map[string]interface{}{
		"records": records,
		"total":   total,
	})
}

func (s *Server) handleApprovalWhitelist(w http.ResponseWriter, r *http.Request) {
	mgr := s.getApprovalManager()
	if mgr == nil {
		jsonResponse(w, map[string]interface{}{"error": "approval system not available"})
		return
	}

	switch r.Method {
	case http.MethodGet:
		patterns := mgr.GetWhitelist()
		jsonResponse(w, map[string]interface{}{
			"patterns": patterns,
			"total":    len(patterns),
		})

	case http.MethodPost:
		var req struct {
			Pattern string `json:"pattern"`
		}
		if err := json.NewDecoder(r.Body).Decode(&req); err != nil || req.Pattern == "" {
			http.Error(w, "Invalid request body", http.StatusBadRequest)
			return
		}
		if err := mgr.AddToWhitelist(req.Pattern); err != nil {
			http.Error(w, err.Error(), http.StatusBadRequest)
			return
		}
		jsonResponse(w, map[string]bool{"success": true})

	case http.MethodDelete:
		// 同时支持 query param 和 body，便于不同客户端调用
		pattern := r.URL.Query().Get("pattern")
		if pattern == "" {
			var req struct {
				Pattern string `json:"pattern"`
			}
			if err := json.NewDecoder(r.Body).Decode(&req); err != nil || req.Pattern == "" {
				http.Error(w, "Invalid request body", http.StatusBadRequest)
				return
			}
			pattern = req.Pattern
		}
		if err := mgr.RemoveFromWhitelist(pattern); err != nil {
			http.Error(w, err.Error(), http.StatusInternalServerError)
			return
		}
		jsonResponse(w, map[string]bool{"success": true})

	default:
		http.Error(w, "Method not allowed", http.StatusMethodNotAllowed)
	}
}

func (s *Server) handleApprovalPatternsDenied(w http.ResponseWriter, r *http.Request) {
	mgr := s.getApprovalManager()
	if mgr == nil {
		jsonResponse(w, map[string]interface{}{"error": "approval system not available"})
		return
	}

	switch r.Method {
	case http.MethodGet:
		denied := mgr.GetDeniedCommands()
		jsonResponse(w, map[string]interface{}{
			"patterns": denied,
			"total":    len(denied),
		})

	case http.MethodDelete:
		// 同时支持 query param 和 body
		pattern := r.URL.Query().Get("pattern")
		if pattern == "" {
			var req struct {
				Pattern string `json:"pattern"`
			}
			if err := json.NewDecoder(r.Body).Decode(&req); err != nil || req.Pattern == "" {
				http.Error(w, "Invalid request body", http.StatusBadRequest)
				return
			}
			pattern = req.Pattern
		}
		// 直接删除该 pattern 记录，避免 Approve 副作用（如新增 trusted pattern、记录 history）。
		// 之前的实现用 Approve 来"抵消" denied，反而会把命令加入 trusted，造成安全风险。
		if err := mgr.RemovePattern(pattern); err != nil {
			http.Error(w, err.Error(), http.StatusInternalServerError)
			return
		}
		jsonResponse(w, map[string]bool{"success": true})

	default:
		http.Error(w, "Method not allowed", http.StatusMethodNotAllowed)
	}
}

func (s *Server) handleApprovalPatternsTrusted(w http.ResponseWriter, r *http.Request) {
	mgr := s.getApprovalManager()
	if mgr == nil {
		jsonResponse(w, map[string]interface{}{"error": "approval system not available"})
		return
	}

	switch r.Method {
	case http.MethodGet:
		trusted := mgr.GetTrustedCommands()
		jsonResponse(w, map[string]interface{}{
			"patterns": trusted,
			"total":    len(trusted),
		})

	case http.MethodDelete:
		// 同时支持 query param 和 body
		pattern := r.URL.Query().Get("pattern")
		if pattern == "" {
			var req struct {
				Pattern string `json:"pattern"`
			}
			if err := json.NewDecoder(r.Body).Decode(&req); err != nil || req.Pattern == "" {
				http.Error(w, "Invalid request body", http.StatusBadRequest)
				return
			}
			pattern = req.Pattern
		}
		// 直接删除该 pattern 记录，避免 Deny 副作用（新增 denied pattern + history 记录）。
		if err := mgr.RemovePattern(pattern); err != nil {
			http.Error(w, err.Error(), http.StatusInternalServerError)
			return
		}
		jsonResponse(w, map[string]bool{"success": true})

	default:
		http.Error(w, "Method not allowed", http.StatusMethodNotAllowed)
	}
}

// isValidStrategy 校验 strategy 是否合法。
func isValidStrategy(s string) bool {
	switch s {
	case "manual", "auto", "smart", "whitelist":
		return true
	}
	return false
}

func (s *Server) handleApprovalStrategy(w http.ResponseWriter, r *http.Request) {
	mgr := s.getApprovalManager()
	if mgr == nil {
		jsonResponse(w, map[string]interface{}{"error": "approval system not available"})
		return
	}

	if r.Method == http.MethodPost {
		var req struct {
			Strategy string `json:"strategy"`
		}
		if err := json.NewDecoder(r.Body).Decode(&req); err != nil || req.Strategy == "" {
			http.Error(w, "Invalid request body", http.StatusBadRequest)
			return
		}

		if !isValidStrategy(req.Strategy) {
			http.Error(w, "Invalid strategy. Must be: manual, auto, smart, or whitelist", http.StatusBadRequest)
			return
		}
		mgr.SetStrategy(approval.Strategy(req.Strategy))

		// 持久化到主配置文件，避免重启丢失
		s.syncApprovalToMainConfig(mgr)

		jsonResponse(w, map[string]bool{"success": true})
		return
	}

	// GET current strategy
	cfg := mgr.GetConfig()
	jsonResponse(w, map[string]interface{}{
		"strategy": cfg.Strategy,
	})
}

func (s *Server) handleApprovalClearHistory(w http.ResponseWriter, r *http.Request) {
	mgr := s.getApprovalManager()
	if mgr == nil {
		jsonResponse(w, map[string]interface{}{"error": "approval system not available"})
		return
	}

	var req struct {
		OlderThanHours int `json:"older_than_hours"`
	}
	// Default body is optional; use default 168 hours (7 days) if not provided
	_ = json.NewDecoder(r.Body).Decode(&req)
	if req.OlderThanHours <= 0 {
		req.OlderThanHours = 168
	}

	mgr.ClearHistory(time.Duration(req.OlderThanHours) * time.Hour)
	jsonResponse(w, map[string]bool{"success": true})
}

func (s *Server) getApprovalManager() *approval.Manager {
	// Prefer the independent approval manager (always available after server init)
	if s.approvalMgr != nil {
		return s.approvalMgr
	}

	// Fallback: try to find any agent with an approval hook
	s.agentsMu.Lock()
	defer s.agentsMu.Unlock()
	for _, a := range s.agents {
		if hook := a.GetApprovalHook(); hook != nil {
			return hook.GetManager()
		}
	}

	return nil
}

func (s *Server) handleApprovalPending(w http.ResponseWriter, r *http.Request) {
	mgr := s.getApprovalManager()
	if mgr == nil {
		jsonResponse(w, map[string]interface{}{"error": "approval system not available"})
		return
	}

	pending := mgr.GetPendingApprovals()

	type pendingItem struct {
		ID        string `json:"id"`
		Command   string `json:"command"`
		RiskLevel string `json:"risk_level"`
		SessionID string `json:"session_id"`
		CreatedAt string `json:"created_at"`
		ExpiresAt string `json:"expires_at"`
	}

	items := make([]pendingItem, 0, len(pending))
	for _, pa := range pending {
		items = append(items, pendingItem{
			ID:        pa.ID,
			Command:   pa.Request.Command,
			RiskLevel: pa.Request.RiskLevel.String(),
			SessionID: pa.Request.SessionID,
			CreatedAt: pa.CreatedAt.Format(time.RFC3339),
			ExpiresAt: pa.ExpiresAt.Format(time.RFC3339),
		})
	}

	jsonResponse(w, map[string]interface{}{
		"pending": items,
		"total":   len(items),
	})
}

func (s *Server) handleApprovalDenied(w http.ResponseWriter, r *http.Request) {
	mgr := s.getApprovalManager()
	if mgr == nil {
		jsonResponse(w, map[string]interface{}{"error": "approval system not available"})
		return
	}

	denied := mgr.GetDeniedCommands()
	jsonResponse(w, map[string]interface{}{
		"patterns": denied,
		"total":    len(denied),
	})
}

// handleApprovalSettings 同时承担 GET（返回当前设置）和 PUT（更新设置）。
// GET 路径与 handleApprovalStatus 行为一致，二者路由不同但返回字段相同。
func (s *Server) handleApprovalSettings(w http.ResponseWriter, r *http.Request) {
	mgr := s.getApprovalManager()
	if mgr == nil {
		jsonResponse(w, map[string]interface{}{"error": "approval system not available"})
		return
	}

	switch r.Method {
	case http.MethodGet:
		s.respondApprovalSettings(w, mgr)
	case http.MethodPut:
		var req struct {
			Strategy       string `json:"strategy"`
			TrustThreshold int    `json:"trust_threshold"`
			CLIPrompt      bool   `json:"cli_confirm"`
			EnableLearning bool   `json:"enable_learning"`
		}
		if err := json.NewDecoder(r.Body).Decode(&req); err != nil {
			http.Error(w, "invalid request body", http.StatusBadRequest)
			return
		}
		// 校验 strategy 合法性（空字符串表示不更新）
		if req.Strategy != "" && !isValidStrategy(req.Strategy) {
			http.Error(w, "Invalid strategy. Must be: manual, auto, smart, or whitelist", http.StatusBadRequest)
			return
		}

		// 使用 SetXxx 方法更新，避免修改 GetConfig() 返回的局部拷贝（GetConfig 现在返回值类型）。
		if req.Strategy != "" {
			mgr.SetStrategy(approval.Strategy(req.Strategy))
		}
		if req.TrustThreshold > 0 {
			mgr.SetTrustThreshold(req.TrustThreshold)
		}
		mgr.SetEnableCLIConfirm(req.CLIPrompt)
		mgr.SetEnableLearning(req.EnableLearning)

		// 持久化到主配置文件
		s.syncApprovalToMainConfig(mgr)

		jsonResponse(w, map[string]bool{"success": true})
	default:
		http.Error(w, "method not allowed", http.StatusMethodNotAllowed)
	}
}

// respondApprovalSettings 统一构造 settings/status 响应体，避免两处字段不一致。
// JSON 字段统一使用 snake_case，与 ApprovalRecord 等其它接口保持一致。
func (s *Server) respondApprovalSettings(w http.ResponseWriter, mgr *approval.Manager) {
	cfg := mgr.GetConfig()
	stats := mgr.GetStats()
	jsonResponse(w, map[string]interface{}{
		"strategy":             cfg.Strategy,
		"enable_learning":      cfg.EnableLearning,
		"cli_confirm":          cfg.EnableCLIConfirm,
		"trust_threshold":      cfg.TrustThreshold,
		"approval_timeout":     cfg.ApprovalTimeout,
		"enable_whitelist":     cfg.EnableWhitelist,
		"whitelist":            mgr.GetWhitelist(),
		"trusted_patterns":     stats.TrustedPatterns,
		"denied_patterns":      stats.DeniedPatterns,
		"total_requests":       stats.TotalRequests,
		"auto_approved":        stats.AutoApproved,
		"user_approved":        stats.UserApproved,
		"user_denied":          stats.UserDenied,
		"avg_response_time_ms": stats.AvgResponseTime,
	})
}

func (s *Server) handleApprovalTrusted(w http.ResponseWriter, r *http.Request) {
	mgr := s.getApprovalManager()
	if mgr == nil {
		jsonResponse(w, map[string]interface{}{"error": "approval system not available"})
		return
	}

	trusted := mgr.GetTrustedCommands()
	jsonResponse(w, map[string]interface{}{
		"patterns": trusted,
		"total":    len(trusted),
	})
}

func (s *Server) handleApprovalStats(w http.ResponseWriter, r *http.Request) {
	mgr := s.getApprovalManager()
	if mgr == nil {
		jsonResponse(w, map[string]interface{}{"error": "approval system not available"})
		return
	}

	stats := mgr.GetStats()

	// Convert by_risk_level keys to strings for JSON serialization
	byRiskLevel := make(map[string]int)
	for k, v := range stats.ByRiskLevel {
		byRiskLevel[k.String()] = v
	}

	jsonResponse(w, map[string]interface{}{
		"total_requests":       stats.TotalRequests,
		"auto_approved":        stats.AutoApproved,
		"user_approved":        stats.UserApproved,
		"user_denied":          stats.UserDenied,
		"timed_out":            stats.TimedOut,
		"trusted_patterns":     stats.TrustedPatterns,
		"denied_patterns":      stats.DeniedPatterns,
		"top_commands":         stats.TopCommands,
		"by_risk_level":        byRiskLevel,
		"by_category":          stats.ByCategory,
		"avg_response_time_ms": stats.AvgResponseTime,
	})
}

// syncApprovalToMainConfig 将 approval.Manager 中的配置回写到主配置文件，
// 使 strategy / trust_threshold / enable_learning / cli_confirm / approval_timeout
// 等可在重启后保留。其它字段（dangerous_patterns 等）由 DefaultConfig 维护，不在此同步。
func (s *Server) syncApprovalToMainConfig(mgr *approval.Manager) {
	if s.cfg == nil {
		return
	}
	ac := mgr.GetConfig()
	s.cfg.Approval = &appconfig.ApprovalConfig{
		Strategy:         string(ac.Strategy),
		TrustThreshold:   ac.TrustThreshold,
		EnableLearning:   ac.EnableLearning,
		EnableCLIConfirm: ac.EnableCLIConfirm,
		ApprovalTimeout:  ac.ApprovalTimeout,
	}
	_ = s.cfg.Save()
}

func (s *Server) handleApprovalPendingByID(w http.ResponseWriter, r *http.Request) {
	mgr := s.getApprovalManager()
	if mgr == nil {
		jsonResponse(w, map[string]interface{}{"error": "approval system not available"})
		return
	}

	// Extract ID from path: /api/approval/pending/{id}/resolve or /api/approval/pending/{id}
	path := strings.TrimPrefix(r.URL.Path, "/api/approval/pending/")
	parts := strings.SplitN(path, "/", 2)
	id := parts[0]

	if r.Method == http.MethodPost && len(parts) > 1 && parts[1] == "resolve" {
		var req struct {
			Approved bool   `json:"approved"`
			Reason   string `json:"reason"`
		}
		if err := json.NewDecoder(r.Body).Decode(&req); err != nil {
			http.Error(w, "Invalid request body", http.StatusBadRequest)
			return
		}

		// ResolveWebApproval 现在返回 error，需要根据错误类型返回合适的状态码。
		if err := mgr.ResolveWebApproval(id, req.Approved, req.Reason); err != nil {
			// 区分 not found（404）与 expired（410）；其它统一 500
			if strings.Contains(err.Error(), "not found") {
				http.Error(w, err.Error(), http.StatusNotFound)
			} else if strings.Contains(err.Error(), "expired") {
				http.Error(w, err.Error(), http.StatusGone)
			} else {
				http.Error(w, err.Error(), http.StatusInternalServerError)
			}
			return
		}
		jsonResponse(w, map[string]bool{"success": true})
		return
	}

	// GET single pending approval
	pending := mgr.GetPendingApprovals()
	for _, pa := range pending {
		if pa.ID == id {
			jsonResponse(w, map[string]interface{}{
				"id":         pa.ID,
				"command":    pa.Request.Command,
				"risk_level": pa.Request.RiskLevel.String(),
				"session_id": pa.Request.SessionID,
				"created_at": pa.CreatedAt.Format(time.RFC3339),
				"expires_at": pa.ExpiresAt.Format(time.RFC3339),
			})
			return
		}
	}

	http.Error(w, "Pending approval not found", http.StatusNotFound)
}

// handleApprovalStatus 返回当前审批系统状态（与 settings GET 等价，保留独立路由以兼容前端）。
func (s *Server) handleApprovalStatus(w http.ResponseWriter, r *http.Request) {
	mgr := s.getApprovalManager()
	if mgr == nil {
		jsonResponse(w, map[string]interface{}{"error": "approval system not available"})
		return
	}
	s.respondApprovalSettings(w, mgr)
}
